packages/ripple-binary-codec/src/types/serialized-type.ts

```diff
@@ -80,26 +80,30 @@ class SerializedType {
 }
 
 /**
- * Base class for SerializedTypes that are comparable
+ * Base class for SerializedTypes that are comparable.
+ *
+ * @template T - What types you want to allow comparisons between. You must specify all types.
```

Comment:
I think this extra sentence helps explain why we're using type generics more explicitly 
```suggestion
 * @template T - What types you want to allow comparisons between. You must specify all types. Primarily used to allow comparisons between built-in types (like `string`) and SerializedType subclasses (like `Hash`). 
```